dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityCacheTest.java

```diff
@@ -129,6 +146,31 @@ void shouldFindTrackedEntityWhenAnotherOwnerAlreadyInCache()
     Assertions.assertContainsOnly(List.of(trackedEntityC), trackedEntities);
   }
 
+  @Test
+  void shouldHaveAccessAfterEnrollmentPersistedAndOwnershipCreatedAndCacheInvalidated() {
+    TrackedEntity trackedEntity = manager.get(TrackedEntity.class, "H8732208127");
+    assertNotNull(trackedEntity);
+    User user = manager.get(User.class, "lPaILkLkgOM");
+
+    injectSecurityContextUser(user);
+
+    assertFalse(
+        trackerOwnershipManager.hasAccess(
+            UserDetails.fromUser(regularUser), trackedEntity, programA));
+
+    Enrollment enrollment =
+        createEnrollment(trackedEntity.getUid(), programA.getUid(), "h4w96yEMlzO");
```

Comment:
Would it make sense to extract `h4w96yEMlzO` to a variable and give it a meaningful name?